### PR TITLE
CORE-7309 Api to find my signing keys

### PR DESCRIPTION
**Pull request by kyriathar**
- implements find my signing keys API on `SigningServiceImpl`. Delegates to existing underlying infrastructure to find owned keys and then creates a mapping from requested key to found to be owned key or null if key is not owned by caller. In case of composite keys it maps the composite key to the first found key leaf.
- converts requested keys to their IDs to create Kafka request containing the key IDs instead serialized keys.

api PR: [#794](https://github.com/corda/corda-api/pull/794)

### CODE REVIEW

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/application/crypto/SigningServiceImpl.kt [hunk 1]
```diff
@@ -34,4 +36,44 @@ class SigningServiceImpl @Activate constructor(
...
+        }
+
+        val compositeKeysReqResp = compositeKeys.associateWith {
+            var foundLeaf: PublicKey? = null
+            it.leafKeys.forEach { leaf ->
+                if (leaf in foundSigningKeys) {
+                    if (foundLeaf == null) {
+                        foundLeaf = leaf
+                    } else {
+                        throw IllegalStateException(
```
Comment (kyriathar) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/application/crypto/SigningServiceImpl.kt]: How should we behave when more than one leaves are found to be owned by the same node for the same composite key? @mnesbit could you advice please? Currently throwing, should we change that to info log it and just proceed with mapping composite key with first found leaf?

Comment (kyriathar) [reply components/flow/flow-service/src/main/kotlin/net/corda/flow/application/crypto/SigningServiceImpl.kt]: Agreed offline to map a composite key to the first found leaf owned by the vnode and ignore other found leaves logging a message.

**Review by mnesbit — DISMISSED** [review]
OK

**Review by mnesbit — APPROVED** [review]
OK

**Pull request merged**